handlers.js

```diff
@@ -415,28 +916,132 @@
+                function sendFileList() {
+                        results[w].getMetadata(function(file) {
+                            //console.log(file)
+                            var rawname = results[w].name
+                            //from https://stackoverflow.com/questions/10420352/converting-file-size-in-bytes-to-human-readable-string/10420404
+                            function humanFileSize(bytes, si=false, dp=1) {
+                              const thresh = si ? 1000 : 1024;
 
-            this.setHeader('transfer-encoding','chunked')
-            this.writeHeaders(200)
-            this.writeChunk(WSC.template_data )
-            var html = ['<script>start("current directory...")</script>',
-                        '<script>addRow("..","..",1,"170 B","10/2/15, 8:32:45 PM");</script>']
+                              if (Math.abs(bytes) < thresh) {
+                                return bytes + ' B';
+                              }
 
-            for (var i=0; i<results.length; i++) {
-                var rawname = results[i].name
-                var name = encodeURIComponent(results[i].name)
-                var isdirectory = results[i].isDirectory
-                var filesize = '""'
-                //var modified = '10/13/15, 10:38:40 AM'
-                var modified = ''
-                // raw, urlencoded, isdirectory, size, 
-                html.push('<script>addRow("'+rawname+'","'+name+'",'+isdirectory+','+filesize+',"'+modified+'");</script>')
-            }
-            var data = html.join('\n')
-            data = new TextEncoder('utf-8').encode(data).buffer
-            this.writeChunk(data)
-            this.request.connection.write(WSC.str2ab('0\r\n\r\n'))
-            this.finish()
+                              const units = si 
+                                ? ['kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'] 
+                                : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+                              let u = -1;
+                              const r = 10**dp;
+
+                              do {
+                                bytes /= thresh;
+                                ++u;
+                              } while (Math.round(Math.abs(bytes) * r) / r >= thresh && u < units.length - 1);
+
+
+                              return bytes.toFixed(dp) + ' ' + units[u];
+                            }
+
+                            var lastModifiedMonth = file.modificationTime.getMonth() + 1
+                            var lastModifiedDay = file.modificationTime.getDate()
+                            var lastModifiedYear = file.modificationTime.getFullYear().toString().substring(2, 4)
+                            var lastModifiedHours = file.modificationTime.getHours() + 1
+                            var lastModifiedMinutes = file.modificationTime.getMinutes() + 1
+                            var lastModifiedSeconds = file.modificationTime.getSeconds() + 1
```

Comment:
are you sure you add 1 to seconds/minutes? I think you only have to do that for month? looks weird..

Reply:
I'll look at this

Reply:
Fixed